nginx/Dockerfile

```diff
@@ -37,16 +38,14 @@ RUN ./configure \
 	--with-http_ssl_module \
 	--with-openssl=/tmp/openssl-${openssl_version} \
 	--with-openssl-opt="enable-weak-ssl-ciphers" \
-	--with-pcre
+	--without-http_rewrite_module
 RUN make -j$(nproc)
 RUN make install
 
-# * Use Ubuntu as we need to install several pieces of software
-FROM ubuntu:latest
+# * Use Ubuntu to run Nginx
+FROM ubuntu:latest AS final
```

Comment:
It's probably best if we don't use `:latest` in images meant for production use. Use a specific version instead (like is done a few lines up)

Reply:
Which version? Latest LTS?

Reply:
Same one as used above so the client doesn't have to download yet another Ubuntu layer.

It's just so it doesn't randomly break when a new Ubuntu version releases.

Reply:
The build version uses Ubuntu 18, which has long since left support and is no longer receiving security updates. It's only used as the build image since it's known to ship with support for the older TLS versions and SSL ciphers we need.

Is it really a good idea to actually use it as the running, accessible, image though? Given that it's no longer being supported?

Reply:
I should note that I haven't done extensive testing with other versions as the build image. Newer ones could work, either out of the box or with some extra dependencies. I only used this version since it's known to work, which reduced dev time.

Maybe it's worth exploring if other versions also build correctly, if downloading more images is a concern?

Reply:
Are those tls things not needed in runtime?

Reply:
Since latest was used as final stage before the refactor. Latest LTS should just work, so Id go with that.

I don't think it's worth the time investment of trying more versions. It doesn't impact performance other than a bit more storage space. We just need the image not to break when an Ubuntu update releases by specifying a fixed version.

Reply:
> Are those tls things not needed in runtime?

They are, but not necessarily by the OS. Just by nginx/OpenSSL. But trying to install nginx normally on Ubuntu 20+ won't have support for the TLS/SSL we need. That's why nginx/OpenSSL is installed and built from source on a system designed to support those versions, and then is moved to the final image.

Ubuntu dropping support for TLS 1.0/1.1 after 18 is the reason we switched away from nginx entirely in favor of CF Tunnels, and why Juxt was using Ubuntu 18 containers for a while before that.

So long as nginx/OpenSSL can support the TLS versions and SSL ciphers we need then it doesn't matter what the OS version is tbh.

Reply:
> Since latest was used as final stage before the refactor. Latest LTS should just work, so Id go with that.
> 
> I don't think it's worth the time investment of trying more versions. It doesn't impact performance other than a bit more storage space. We just need the image not to break when an Ubuntu update releases by specifying a fixed version.

That sounds reasonable. Maybe we should lock it to the current LTS version? I believe that's 22?

Reply:
> it's known to ship with support for the older TLS versions and SSL ciphers we need

Why is this necessary if we compile OpenSSL ourselves?

Reply:
Looks like it works fine with the latest Ubuntu LTS, 24.04.